### DISCUSSION TITLE
Please support pydantic-2.x

### DISCUSSION THREAD

**Discussion by yurivict** [fastapi/fastapi#10873] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
n/a
```

### Description

PYPI now has pydantic at 2.x and having two versions of Pydantic at the same time causes conflicts in resolving dependencies for some other packages, because pydantic-1.x and pydantic-2.x use the same namespace 'pydantic'.

I am trying to create the FreeBSD port for FreqTrade https://www.freqtrade.io/, but it requires pydantic-2.x, but FastAPI is one of its dependencies, and it requires pydantic-1.x

2 versions of the same package can't exist simultaneously in the same environment.

### Operating System

Other

### Operating System Details

FreeBSD

### FastAPI Version

0.108.0

### Pydantic Version

1.x

### Python Version

3.9

### Additional Context

There is [another discussion](https://github.com/tiangolo/fastapi/issues/10360) mentioning pydantic with unclear outcome.

**Comment by n8sty** [comment]
The version of FastAPI that you're using supports using Pydantic 2.x https://github.com/tiangolo/fastapi/blob/040ad986d48bb9a400de804f7f25abb856d85e1a/pyproject.toml#L44

**Comment by JavierSanchezCastro** [comment]
Since version 0.100.0 FastAPI supports Pydantic v2, please check https://fastapi.tiangolo.com/release-notes/#01000

**Comment by yurivict** [comment]
Thank you.
